clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -148,8 +148,12 @@ public class AzureComputeClient extends AzureBaseClient {
       sg.lastReadTime = System.currentTimeMillis()
       return sg
     } catch (CloudException e) {
-      // treat exception as a http 404 return (resource not found)
-      log.warn("ServerGroup: ${e.message} (${serverGroupName} was not found)")
+      if (e.body.code == 404) {
+        log.warn("ServerGroup: ${e.message} (${serverGroupName} was not found)")
+      }
+      else {
+        throw e
+      }
```

Comment:
Are we 100% sure we get a 404 CloudException if the resource is not there? I've observed different behavior when trying to delete a non-existing resource in some cases; basically there was no exception.

I'm asking because I'd like to take this code and add it into how we handle the delete of an Azure resource.